catroid/src/main/java/org/catrobat/catroid/content/actions/ShowTextAction.kt

```diff
@@ -0,0 +1,135 @@
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2022 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions
+
+import android.util.Log
+import com.badlogic.gdx.scenes.scene2d.Actor
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction
+import com.badlogic.gdx.utils.Array
+import org.catrobat.catroid.content.Scope
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.formulaeditor.UserVariable
+import org.catrobat.catroid.stage.ShowTextActor
+import org.catrobat.catroid.stage.StageActivity
+import org.catrobat.catroid.utils.ShowTextUtils.AndroidStringProvider
+
+class ShowTextAction : TemporalAction() {
+    private var xPosition: Formula? = null
+    private var yPosition: Formula? = null
+    private var variableToShow: UserVariable? = null
+    private var scope: Scope? = null
+    private var androidStringProvider: AndroidStringProvider? = null
+    private val showTextActor: ShowTextActor? = null
+    override fun begin() {
+        try {
+            val xPosition = xPosition?.interpretInteger(scope)
+            val yPosition = yPosition?.interpretInteger(scope)
+            if (StageActivity.stageListener != null) {
+                val stageActors = StageActivity.stageListener.stage.actors
+                iterateActors(stageActors)
+                actor = ShowTextActor(
```

Comment:
Is there a reason why the `ShowTextActor`-object is set to `actor` instead of the field `showTextActor`? Because the position of latter is then updated in the `update()` method, but since it's never set it doesn't have an effect.